### PR TITLE
Override hostname

### PR DESCRIPTION
**Pull request by naohiro-t**
See: https://github.com/scoutapp/scout_apm_python/issues/248
This also requires the binary agent changes, so do not merge yet.

### CODE REVIEW

File: src/scout_apm/core/commands.py [hunk 1]
```diff
@@ -16,13 +16,18 @@ class Register(object):
...
-        logger.info("Registering with app=%s key=%s" % (self.app, self.key))
+        logger.info(
+            "Registering with app=%s key=%s host=%s"
+            % (self.app, self.key, self.hostname)
+        )
         return {
             "Register": {
                 "app": self.app,
                 "key": self.key,
+                "host": self.hostname,
```
Comment (cschneid) [inline src/scout_apm/core/commands.py]: The hostname arg needs to be optional. We should have `None` if it's not set

Comment (naohiro-t) [reply src/scout_apm/core/commands.py]: Should we set default value to `None`?

Comment (adamchainz) [inline src/scout_apm/core/commands.py]: Do you want to pass `host` or `hostname` to the core agent?

Comment (adamchainz) [reply src/scout_apm/core/commands.py]: @cschneid verified on Slack that the agent expects `host`

**Pull request merged**